ibis/backends/pandas/aggcontext.py

```diff
@@ -363,7 +363,13 @@ def agg(self, grouped_data, function, *args, **kwargs):
                 'Object {} is not callable or a string'.format(function)
             )
 
-        return grouped_data.agg(wrap_for_agg(function, args, kwargs))
+        if isinstance(grouped_data, pd.core.groupby.generic.SeriesGroupBy):
+            # Using `apply` as a workaround for the fact that
+            # `SeriesGroupBy.agg` does not allow np.arrays to be returned
+            # from UDFs.
+            return grouped_data.apply(wrap_for_agg(function, args, kwargs))
```

Comment:
In what case we want to exercise the "else" branch here?

Reply:
This is when exercised when doing an ungrouped aggregation, e.g. `my_arr_col.collect()`